### PR TITLE
Make FIFO files more secure

### LINKED ISSUE #2210: Remaining work to close feature #2136
> estimate 1
Parent feature: #2136 

Done criteria:

- [x] Public facing documentation has been added describing the
      feature (man page, or sphinx web page update)
- [x] Documentation has been added to the service web page for service
      related features

### PR DESCRIPTION
**Pull request by avilcheslopez**
- FIFO files are now created in geopm's runtime directory to avoid
security vulnerabilities related to using the tmp directory
- Documentation and tests updated accordingly

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

- Relates to #2136 
- Fixes #2210

Move FIFO files to geopm's runtime directory to make them more secure.

### CODE REVIEW

File: service/docs/source/security.rst [hunk 1]
```diff
@@ -472,11 +472,19 @@ When making directories, if the path already exists checks are
...
+indicate it is invalid and preserved for later auditing. In this case
+a new directory will be created at the specified path. If the path did
+not already exist, a new directory is created with the proper
+permissions.
+
+By default, directories are created with 0o700 permissions (i.e. rwx
+only for the owner). Some directories, for example
+``/var/run/geopm-service``, also require execution permissions (i.e.
+0o711). For more details on how directories are created and default
+permissions, please see the `systme_files.py <http://geopm.github.io/geopmdpy.7.html#module-geopmdpy.system_files>`__
```
Comment (cmcantalupo) [inline service/docs/source/security.rst]:
```suggestion
permissions, please see the `system_files.py <http://geopm.github.io/geopmdpy.7.html#module-geopmdpy.system_files>`__
```

File: service/geopmdpy_test/TestActiveSessions.py [hunk 2]
```diff
@@ -168,7 +169,9 @@ def check_json_file(self, name, contents, is_valid):
              mock.patch('uuid.uuid4', return_value='uuid4'), \
              mock.patch('sys.stderr.write', return_value=None) as mock_err:
             act_sess = ActiveSessions(sess_path)
-            mock_smd.assert_called_once_with(sess_path)
+            mock_smd.assert_called_once_with(
+                sess_path,
+                perm_mode=GEOPM_SERVICE_VAR_PATH_PERM)
```
Comment (cmcantalupo) [inline service/geopmdpy_test/TestActiveSessions.py:174]:
```suggestion
            mock_smd.assert_called_once_with(sess_path,
                perm_mode=GEOPM_SERVICE_VAR_PATH_PERM)
```

File: service/docs/source/security.rst (hunk 1 above)
Comment (avilcheslopez) [reply service/docs/source/security.rst]: Fixed.

File: service/src/BatchStatus.hpp [hunk 3]
```diff
@@ -92,6 +91,11 @@ namespace geopm
             void check_return(int ret, const std::string &func_name);
             int m_read_fd;
             int m_write_fd;
+
+            // This is the single place where the server prefix is located,
+            // which is also accessed by BatchStatusTest.
+            static constexpr const char* M_DEFAULT_FIFO_PREFIX =
+                "/var/run/geopm-service/geopm-service-batch-status-";
```
Comment (cmcantalupo) [inline service/src/BatchStatus.hpp]:
```suggestion
                "/var/run/geopm-service/batch-status-";
```

File: service/test/BatchStatusTest.cpp [hunk 4]
```diff
@@ -59,7 +65,7 @@ class BatchStatusTest : public ::testing::Test
 
 void BatchStatusTest::SetUp(void)
 {
-    m_server_prefix = BatchStatus::M_FIFO_PREFIX;
+    m_server_prefix = "/tmp/geopm-service-batch-status-";
```
Comment (cmcantalupo) [inline service/test/BatchStatusTest.cpp]:
```suggestion
    m_server_prefix = "/tmp/geopm-test-service-batch-status-";
```

**Review by cmcantalupo — COMMENTED** [review]
Just a couple of name change suggestions on file paths.  Otherwise looks great!  Thanks for hiding the fifo path from the public interface.

File: service/geopmdpy_test/TestSecureFiles.py [hunk 5]
```diff
@@ -106,6 +106,19 @@ def test_default_creation(self):
         secure_make_dirs(sess_path)
         self.check_dir_perms(sess_path)
 
+    def test_creation_with_perm(self):
```
Comment (cmcantalupo) [inline service/geopmdpy_test/TestSecureFiles.py:109]: You'll need to add this test fixture name to the Makefile.mk in this directory to have it run with `make check`

**Review by cmcantalupo — APPROVED** [review]
👍

**Pull request merged**